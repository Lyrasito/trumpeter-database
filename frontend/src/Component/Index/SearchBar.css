.searchbar {
  display: flex;
  justify-content: space-between;
  align-items: left;
  width: 80%;
  margin-bottom: 20px;
}

.container {
  height: 230px;
  width: 1280px;
  margin-left: 80px;
}

.allPlayers {
  text-decoration: underline;
  cursor: pointer;
}
.search {
  box-sizing: border-box;
  height: 63px;
  width: 300px;
  border: 3px solid #666666;
  border-radius: 30px;
  background-color: #ffffff;
  display: block;
  margin-top: 10px;
  font-size: 24px;
}

.search:focus {
  border: 3px solid #ffc600;
}

#nameSearch {
  box-sizing: border-box;
  height: 63px;
  width: 767px;
  border: 3px solid #666666;
  border-radius: 30px;
  background-color: #ffffff;
  margin-bottom: 20px;
  font-size: 24px;
}

#nameSearch:focus {
  border: 3px solid #ffc600;
}

.click-container {
  display: flex;
}
.clicked {
  text-decoration: underline;
  margin-right: 25px;
  pointer-events: none;
}

.clickable {
  color: #aaaaaa;
  cursor: pointer;
  margin-right: 25px;
}
svg {
  width: 20px;
  height: 20px;
  margin-left: 10px;
  margin-bottom: -2px;
}
