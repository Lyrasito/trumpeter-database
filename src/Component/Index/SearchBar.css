.searchbar {
  display: flex;
  justify-content: space-around;
  align-items: center;
  width: 100%;
}

.container {
  text-align: center;
  background-image: url("https://cdn.pixabay.com/photo/2012/04/18/20/36/pipe-37816_960_720.png");
  background-position: center;
  background-color: black;
}
.button-container {
  text-align: center;
}
.submit {
  margin: 50px 0 30px 0;
  border: 2px solid white;
  border-radius: 5px;
  color: white;
  background-color: navy;
  text-align: center;
  width: 100px;
  height: 30px;
  cursor: pointer;
}

.allPlayers {
  margin: 20px 0 50px 0;
  border: 2px solid white;
  border-radius: 5px;
  color: white;
  background-color: navy;
  text-align: center;
  width: 100px;
  height: 50px;
  cursor: pointer;
}

.search {
  height: 30px;
  width: 150px;
  text-align: center;
  margin-top: 40px;
}

#nameSearch {
  margin-top: 50px;
}

.nameSearch {
  text-align: center;
}
