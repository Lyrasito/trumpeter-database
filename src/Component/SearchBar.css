.searchbar {
  display: flex;
  justify-content: space-around;
  align-items: center;
  width: 100%;
}

.container {
  text-align: center;
}
.button-container {
  text-align: center;
  width: 83%;
}
.submit {
  margin: 50px 0 30px 0;
  border: 2px solid white;
  border-radius: 5px;
  color: white;
  background-color: navy;
  text-align: center;
  width: 100px;
  height: 30px;
  cursor: pointer;
}

.search {
  height: 30px;
  width: 150px;
}
