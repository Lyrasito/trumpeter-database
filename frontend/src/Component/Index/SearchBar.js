import React from "react";
import "./SearchBar.css";
import SearchByName from "./SearchByName";
import SearchByCityYearGenre from "./SearchByCityYearGenre";

class SearchBar extends React.Component {
  state = {
    searchResults: [],
    searchByName: true,
    SearchByCityYearGenre: false,
    haveSearched: false,
  };

  allPlayersButton = () => {
    this.setState({
      searchResults: this.props.allPlayers,
    });
  };

  changeSearch = () => {
    this.setState({
      searchByName: !this.state.searchByName,
      searchByCityYearGenre: !this.state.searchByCityYearGenre,
    });
  };

  haveSearched = () => {
    this.setState({ haveSearched: true });
  };
  renderNoResults = () => {
    if (this.props.searchResults.length === 0 && this.state.haveSearched) {
      return <h4>Sorry, your search returned no results.</h4>;
    }
  };

  render() {
    return (
      <div className="container">
        <div className="click-container">
          <h4
            className={this.state.searchByName ? "clicked" : "clickable"}
            onClick={this.changeSearch}
          >
            Search by name
          </h4>
          <h4
            className={
              this.state.SearchByCityYearGenre ? "clicked" : "clickable"
            }
            onClick={this.changeSearch}
          >
            Search by city, year, genre
          </h4>
        </div>
        {this.state.searchByName && (
          <SearchByName
            searchByName={this.props.searchByName}
            haveSearched={this.haveSearched}
          />
        )}
        {this.state.searchByCityYearGenre && (
          <SearchByCityYearGenre
            searchPlayers={this.props.searchPlayers}
            haveSearched={this.haveSearched}
          />
        )}
        <h5 onClick={this.props.allPlayers} className="allPlayers">
          Or, see all players
        </h5>
        {this.renderNoResults()}
      </div>
    );
  }
}

export default SearchBar;
