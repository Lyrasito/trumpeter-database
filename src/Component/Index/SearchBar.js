import React from "react";
import "./SearchBar.css";
import { ReactComponent as SearchLogo } from "../../svg/Search.svg";

class SearchBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      searchResults: [],
      city: "",
      year: "",
      genre: "",
      name: "",
      searchBy: "name",
      haveSearched: false,
    };
    this.allPlayersButton = this.allPlayersButton.bind(this);
    this.handleCityChange = this.handleCityChange.bind(this);
    this.handleYearChange = this.handleYearChange.bind(this);
    this.handleGenreChange = this.handleGenreChange.bind(this);
    this.handleNameChange = this.handleNameChange.bind(this);
    this.search = this.search.bind(this);
    this.nameSearch = this.nameSearch.bind(this);
    this.changeSearch = this.changeSearch.bind(this);
    this.renderNoResults = this.renderNoResults.bind(this);
  }
  allPlayersButton() {
    this.setState({
      searchResults: this.props.allPlayers,
    });
    this.setState({ haveSearched: true });
  }
  handleCityChange(event) {
    this.setState({
      city: event.target.value,
    });
  }

  handleYearChange(event) {
    this.setState({
      year: event.target.value,
    });
  }

  handleGenreChange(event) {
    this.setState({
      genre: event.target.value,
    });
  }

  handleNameChange(event) {
    this.setState({
      name: event.target.value,
    });
  }

  search() {
    this.props.searchPlayers(
      this.state.city,
      this.state.year,
      this.state.genre
    );
    this.setState({ haveSearched: true });
  }
  nameSearch() {
    this.props.searchByName(this.state.name);
    this.setState({ haveSearched: true });
  }
  changeSearch() {
    if (this.state.searchBy === "name") {
      this.setState({
        searchBy: "cityGenreYear",
      });
    } else {
      this.setState({
        searchBy: "name",
      });
    }
  }
  renderNoResults() {
    if (this.state.searchResults.length === 0 && this.state.haveSearched) {
      return <h4>Sorry, your search returned no results.</h4>;
    }
  }

  render() {
    if (this.state.searchBy === "name") {
      return (
        <div className="container">
          <div className="click-container">
            <h4 className="clicked">Search by name</h4>
            <h4 onClick={this.changeSearch} className="clickable">
              Search by genre, year, city
            </h4>
          </div>
          <div className="nameSearch">
            <input id="nameSearch" onChange={this.handleNameChange}></input>
            <br />
            <button className="submit" onClick={this.nameSearch}>
              Search
              <SearchLogo />
            </button>
            <br />
          </div>
          <h5 onClick={this.props.allPlayers} className="allPlayers">
            Or, see all players
          </h5>
          {this.renderNoResults()}
        </div>
      );
    } else if (this.state.searchBy === "cityGenreYear") {
      return (
        <div className="container">
          <div className="click-container">
            <h4 onClick={this.changeSearch} className="clickable">
              Search by name
            </h4>
            <h4 className="clicked">Search by city, year, genre</h4>
          </div>
          <div className="searchbar">
            <label>
              City
              <input
                className="search"
                id="citySearch"
                onChange={this.handleCityChange}
              ></input>
            </label>
            <label>
              Year
              <input
                className="search"
                id="yearSearch"
                onChange={this.handleYearChange}
              ></input>
            </label>
            <label>
              Genre
              <input
                className="search"
                id="genreSearch"
                onChange={this.handleGenreChange}
              ></input>
            </label>
          </div>
          <div className="button-container">
            <button className="submit" onClick={this.search}>
              Search
              <SearchLogo />
            </button>
            <br />
            <h5 onClick={this.props.allPlayers} className="allPlayers">
              Or, see all players
            </h5>
          </div>
          {this.renderNoResults()}
        </div>
      );
    }
  }
}

export default SearchBar;
