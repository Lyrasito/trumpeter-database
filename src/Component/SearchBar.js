import React from "react";
import "./SearchBar.css";

class SearchBar extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      searchResults: [],
      city: "",
      year: "",
      genre: "",
    };
    this.allPlayersButton = this.allPlayersButton.bind(this);
    this.handleCityChange = this.handleCityChange.bind(this);
    this.handleYearChange = this.handleYearChange.bind(this);
    this.handleGenreChange = this.handleGenreChange.bind(this);
    this.search = this.search.bind(this);
  }
  allPlayersButton() {
    this.setState({
      searchResults: this.props.allPlayers,
    });
  }
  handleCityChange(event) {
    this.setState({
      city: event.target.value,
    });
  }

  handleYearChange(event) {
    this.setState({
      year: event.target.value,
    });
  }

  handleGenreChange(event) {
    this.setState({
      genre: event.target.value,
    });
  }
  search() {
    this.props.searchPlayers(
      this.state.city,
      this.state.year,
      this.state.genre
    );
  }

  render() {
    return (
      <div className="container">
        <div className="searchbar">
          <input
            className="search"
            id="citySearch"
            placeholder="Search by City"
            onChange={this.handleCityChange}
          ></input>
          <input
            className="search"
            id="yearSearch"
            placeholder="Search by Year"
            onChange={this.handleYearChange}
          ></input>
          <input
            className="search"
            id="genreSearch"
            placeholder="Search by Genre"
            onChange={this.handleGenreChange}
          ></input>
          <br />
        </div>
        <div className="button-container">
          <button className="submit" onClick={this.search}>
            Submit
          </button>
          <br />
          <button onClick={this.props.allPlayers} className="allPlayers">
            See All Players
          </button>
        </div>
      </div>
    );
  }
}

export default SearchBar;
