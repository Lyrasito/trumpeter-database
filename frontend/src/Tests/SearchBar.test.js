import SearchBar from "../Component/Index/SearchBar";
import React from "react";
import toJSON from "enzyme-to-json";
import { shallow, mount } from "enzyme";

const fakeResults = [
  {
    id: 1,
    name: "Fake Player",
    city: "Fake City",
    startYear: 1234,
    endYear: 5678,
    albums: [
      {
        title: "Fake Album",
        genre: "Fake Genre",
        year: 1234,
      },
    ],
  },
];

describe("<SearchBar />", () => {
  it("renders and matches snapshot", () => {
    const wrapper = shallow(<SearchBar searchResults={fakeResults} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it("changes the search from name to city/year/genre when clicked", () => {
    const wrapper = shallow(<SearchBar searchResults={fakeResults} />);

    expect(wrapper.find("SearchByName")).toHaveLength(1);
    expect(wrapper.find("SearchByCityYearGenre")).toHaveLength(0);

    const searchByCityYearGenre = wrapper.find(
      "[data-test='searchByCityYearGenre']"
    );
    searchByCityYearGenre.simulate("click");
    expect(wrapper.find("SearchByName")).toHaveLength(0);
    expect(wrapper.find("SearchByCityYearGenre")).toHaveLength(1);
  });
});
