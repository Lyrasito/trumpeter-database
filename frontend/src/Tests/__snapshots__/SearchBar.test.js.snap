// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> renders and matches snapshot 1`] = `
<div
  className="container"
>
  <div
    className="click-container"
  >
    <h4
      className="clicked"
      data-test="searchByName"
      onClick={[Function]}
    >
      Search by name
    </h4>
    <h4
      className="clickable"
      data-test="searchByCityYearGenre"
      onClick={[Function]}
    >
      Search by city, year, genre
    </h4>
  </div>
  <SearchByName
    haveSearched={[Function]}
  />
  <h5
    className="allPlayers"
  >
    Or, see all players
  </h5>
</div>
`;
